infrastructure/lambda/api.yaml

```diff
@@ -149,8 +149,32 @@ paths:
         content:
           application/x-www-form-urlencoded:
             schema:
-              $ref: "#/components/schemas/TokenRequest"
-        required: true
+              required:
+                - "grant_type"
+                - "code"
+                - "client_assertion_type"
+                - "client_assertion"
+                - "redirect_uri"
+              properties:
+                grant_type:
+                  type: "string"
+                  pattern: "authorization_code"
+                  example: "authorization_code"
+                code:
+                  type: "string"
+                  minLength: 1
+                client_assertion_type:
+                  type: "string"
+                  pattern: "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"
+                  example: "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"
+                client_assertion:
+                  type: "string"
+                  pattern: "[a-zA-Z0-9_=]+\\.[a-zA-Z0-9_=]+\\.[a-zA-Z0-9_\\-\\+\\/=]+"
+                  example: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0dCIsImlhdCI6MTUxNjIzOTAyMn0.SbcN-ywpLObhMbbaMCtW1Un8LYhQzHsEth9LvTk4oQQ"
+                redirect_uri:
+                  type: "string"
+                  format: "uri"
+                  example: "https://di-ipv-core-stub.london.cloudapps.digital/callback"
```

Comment:
inlining the TokenRequest allows the /token call to pass the API gateway - but I think the validation is not working correctly - @jkunle could you take a look at this in a separate ticket please?